src/sensr_client.cpp

```diff
@@ -58,15 +69,27 @@ namespace sensr
     if (std::find(listeners_.begin(), listeners_.end(), listener) == listeners_.end()) {
       // OutputMessage Port
       if (listener->IsOutputMessageListening()) {
-        ret = output_endpoint_->Connect(GetProtocol() + "://" + address_ + ":5050", 
-        std::bind(&Client::OnResultMessage, this, std::placeholders::_1),
-        std::bind(&Client::OnResultError, this, std::placeholders::_1));
+         if (std::none_of(listeners_.begin(), listeners_.end(), [](const std::shared_ptr<MessageListener>& listener) {
+          return listener->IsOutputMessageListening();
```

Comment:
Can we not reuse `Client::IsResultListening` here?

Reply:
Replaced to if (!IsResultListening()) { now.